daemon/api_snapctl.go

```diff
@@ -96,10 +100,32 @@ func runSnapctl(c *Command, r *http.Request, user *auth.UserState) Response {
 
 	if context != nil && context.IsEphemeral() {
 		context.Lock()
-		defer context.Unlock()
-		if err := context.Done(); err != nil {
+		err := context.Done()
+		context.Unlock()
+		if err != nil {
 			return BadRequest(i18n.G("set failed: %v"), err)
 		}
+
+		chg, err := getRegistryCommitChange(c.d.state, context)
```

Comment:
maybe we should reshap a bit the responsibilities between here and hookstate but not sure